src/json/motion.rs

```diff
@@ -248,11 +248,11 @@ fn json_samples_motion3() {
                 continue;
             }
 
-            let motion = serde_json::from_str::<Motion3>(
+            serde_json::from_str::<Motion3>(
                 &std::fs::read_to_string(&motion)
-                    .expect(&format!("error while reading: {:?}", motion)),
+                    .unwrap_or_else(|_| panic!("error while reading: {:?}", motion)),
```

Comment:
Any particular reason for doing an unwrap_or_else panic instead of expecting?

Reply:
Using `.expect(&format!("..."))` generates the formatted `String` every time when `.expect(...)` is called, which is unwanted. To avoid this, I replaced it with `unwrap_or_else` and a closure, and therefore, the unwanted `String` won't be generated. 

Here is the clippy message:

```console
warning: use of `expect` followed by a function call
   --> src/json/motion.rs:254:14
    |
254 |             .expect(&format!("error while parsing: {:?}", motion));
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|_| panic!("error while parsing: {:?}", motion))`
    |
note: lint level defined here
   --> src/lib.rs:5:5
    |
5   |     clippy::all
    |     ^^^^^^^^^^^
    = note: #[warn(clippy::expect_fun_call)] implied by #[warn(clippy::all)]
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call
```

Reply:
Ah yes, I completely missed that, guess i forgot to run clippy there myself.